utils/distortions.py

```diff
@@ -0,0 +1,791 @@
+
+    Args:
+    x : torch.Tensor
+        Input image tensor with shape `(C, H, W)` where `C` is the number of channels.
+        Pixel values should be in the range `[0, 1]`.
+    levels : int
+        The number of intensity levels to quantize the image into. Must be between 1 and 256.
+
+    Returns:
+    torch.Tensor
+        The quantized image tensor with reduced intensity levels. Pixel values remain in the range `[0, 1]`.
+    """
+    image = kornia.color.rgb_to_grayscale(x) * 255
+    image = image.cpu().numpy()
+    num_classes = levels
+
+    # minimum variance thresholding
+    hist, bins = np.histogram(image, num_classes, [0, 255])
+
+    return_thresholds = np.zeros(num_classes - 1)
+    for i in range(num_classes - 1):
+        return_thresholds[i] = bins[i + 1]
+
+    # quantize image with thresholds
+    bins = torch.tensor([0] + return_thresholds.tolist() + [256])
+    bins = bins.type(torch.int)
+    image = torch.bucketize(x.contiguous() * 255., bins).to(torch.float32)
+    image = mapmm(image)
+    return image
+
+
+def color_block(x: torch.Tensor, pnum: int) -> torch.Tensor:
+    """
+    Adds color blocks to the input image by randomly placing patches of a uniform color.
+
+    Args:
+    x : torch.Tensor
+        Input image tensor with shape `(C, H, W)` where `C` is the number of channels.
+        The pixel values should be in the range `[0, 1]`.
+    pnum : int
+        The number of color blocks (patches) to add to the image.
+
+    Returns:
+    torch.Tensor
+        The image with added color blocks, with the same shape as the input image.
+    """
+    patch_size = [32, 32]
+
+    c, w, h = x.shape
+
+    y = x
+
+    h_max = h - patch_size[0]
+    w_max = w - patch_size[1]
+
+    for i in range(pnum):
+        color = np.random.random(3)
+        px = math.floor(random.random() * w_max)
+        py = math.floor(random.random() * h_max)
+        patch = torch.ones((3, patch_size[0], patch_size[1]))
+        for j in range(3):
+            patch[j, ...] *= color[j]
+        y[:, px:px + patch_size[0], py:py + patch_size[1]] = patch
+
+    return y
+
+
+def high_sharpen(x: torch.Tensor, amount: int, radius: int = 3) -> torch.Tensor:
+    """
+     Applies a high-pass sharpening filter to enhance the high-frequency details of an image.
+
+     Args:
+     x : torch.Tensor
+         Input image tensor in the format `(C, H, W)` where `C` is the number of channels (RGB),
+         with pixel values in the range `[0, 1]`.
+     amount : int
+         The intensity of sharpening to apply. A higher value results in stronger sharpening.
+     radius : int, optional (default=3)
+         The radius of the Gaussian blur used to create the sharpening filter. A larger radius
+         applies a broader blur before the sharpening step.
+
+     Returns:
+     torch.Tensor
+         The sharpened image tensor with the same shape as the input.
+     """
+    x = x[[2, 1, 0], ...]
```

Comment:
Fica mais entendivel: https://pytorch.org/docs/stable/generated/torch.permute.html
```suggestion
    x = torch.permute(x, (2, 1, 0))
```

Reply:
Esse permute vai trocar as dimensões da img, nesse outro tá só convertendo de RGB p/ BGR